lib/reporting/ab_tests_report.rb

```diff
@@ -0,0 +1,95 @@
+# frozen_string_literal: true
+
+begin
+  require 'reporting/cloudwatch_client'
+  require 'reporting/cloudwatch_query_quoting'
+  require 'reporting/command_line_options'
+rescue LoadError => e
+  warn 'could not load paths, try running with "bundle exec rails runner"'
+  raise e
+end
+
+module Reporting
+  class AbTestsReport
+    attr_reader :queries, :time_range
+
+    # @param [Array<String>] queries
+    # @param [Range<Time>] time_range
+    def initialize(
+      queries:,
+      time_range:,
+      verbose: false
+    )
+      @queries = queries
+      @time_range = time_range
+      @verbose = verbose
+    end
+
+    def verbose?
+      @verbose
+    end
+
+    def as_tables
+      queries.map(&method(:table_for_query))
+    end
+
+    def as_emailable_reports
+      queries.map do |query|
+        Reporting::EmailableReport.new(
+          title: query.title,
+          table: table_for_query(query),
+        )
+      end
+    end
+
+    def table_for_query(query)
+      query_data = fetch_results(query: query.query)
+      headers = column_labels(query_data.first)
+      rows = query_data.map(&:values)
+
+      [
+        headers,
+        *format_rows(rows, headers, query.row_labels),
+      ]
+    end
+
+    def format_rows(rows, headers, row_labels)
+      if row_labels
+        rows = rows.each_with_index.map do |value, index|
+          [row_labels[index], *value.slice(1)]
+        end
+      end
+
+      headers.each_index.select { |i| headers[i] =~ /percent/i }.each do |percent_index|
+        rows.each { |row| row[percent_index] = format_as_percent(row[percent_index]) }
+      end
```

Comment:
We have the `float_as_percent` option for `EmailReport` class, so we might be able to leverage that?

https://github.com/18F/identity-idp/blob/main/app/services/reporting/emailable_report.rb#L24

Reply:
I had seen that, though I didn't want to go as far as to assume that any float values returned should be interpreted as a percent value.

Although _maybe_ it'd be safe(r) to make that assumption if we enhanced the logic to check for float values specifically between `0.0` and `1.0` ? 🤔

Reply:
🤷 at the time my thinking was, most numbers thing we show in emails tend to be integers and so the flaots are almost always percents? But yes we have had a mixup with scale of 0-1 vs 0-100 so that seems like it could be a good check